pom.xml

```diff
@@ -0,0 +1,235 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.api</groupId>
+  <artifactId>gax-parent</artifactId>
+  <packaging>pom</packaging>
+  <version>2.18.2-SNAPSHOT</version><!-- {x-version-update:gax:current} -->
+  <name>GAX (Google Api eXtensions) for Java (Parent)</name>
+  <url>https://github.com/googleapis/gax-java</url>
+  <description>Google Api eXtensions for Java (Parent)</description>
+
+  <parent>
+    <groupId>com.google.cloud</groupId>
+    <artifactId>google-cloud-shared-config</artifactId>
+    <version>1.4.0</version>
+  </parent>
+
+  <developers>
+    <developer>
+      <id>GoogleAPIs</id>
+      <name>GoogleAPIs</name>
+      <email>googleapis@googlegroups.com</email>
+      <organization>Google LLC</organization>
+      <organizationUrl>https://www.google.com</organizationUrl>
+    </developer>
+  </developers>
+
+  <scm>
+    <connection>scm:git:https://github.com/googleapis/gax-java.git</connection>
+    <url>https://github.com/googleapis/gax-java</url>
+  </scm>
+
+  <licenses>
+    <license>
+      <name>BSD</name>
+      <url>https://github.com/googleapis/gax-java/blob/master/LICENSE</url>
+    </license>
+  </licenses>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <auto-value.version>1.9</auto-value.version>
+  </properties>
+
+  <modules>
+    <module>gax</module>
+    <module>gax-grpc</module>
+    <module>gax-httpjson</module>
+    <module>gax-bom</module>
+  </modules>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.api</groupId>
+        <artifactId>api-common</artifactId>
+        <version>2.1.5</version>
```

Comment:
This makes inconsistent which versions are used for gradle+bazel and which for maven. I understand that this is a good first step, but for this to work properly, we need some mechanism for a single source of truth for those version numbers between different build systems (now it is dependencies.properties).

Reply:
This is an interim state. The plan is for the Gradle build to go away. We can keep the versions in sync with Bazel manually. Ultimately, we need to figure out if we can just rely on a single build system. Is the Bazel build used for self-service? I'm not sure why we absolutely need it.

Reply:
I agree with Mike as well that eventually we want to be on a single build system, and I checked `gapic-generator-java` and `googleapis` that the bazel stuff in `gax-java` are only used for loading dependencies, which I think can be replaced completely by bazel targets in `gapic-generator-java`. My vision is that the pom.xml in the GAPIC generator mono repo will be the source of truth for dependencies, and I don't think we can get rid of `dependencies.properties` completely before we move `gax-java` to the mono repo.

In short, this is the plan:
1. Add maven build(This PR).
2. Replicate all gradle features(CI, release etc.) in maven and remove gradle build. 
3. Release with Maven dependencies. Before each gax-java release, we may have to manually sync `dependency.properties` to keep things in sync for the time being.
4. Move `gax-java` to the mono repo so that it shares dependency management with `gapic-generator-java`.
5. Do some changes accordingly in the mono repo so that the client library generation depends on the root pom.xml for loading dependencies(This part is not well designed yet but I believe it can be done).

We are probably not going to change the bazel surface in `gapic-generator-java`(like `java_gapic_library`), so that all the client libraries and self service can still be generated seamlessly, but we will change how the dependencies are loaded.

Reply:
@meltsufin Every single automatically generated PR that you currently getting from owlbot is created using bazel build (and that relies on gax dependencies.properties). It is also used for self-service, that is correct.

And it is also very useful for development, though it may look useless on release step. For any gapic-gen/gax development work it s extremely useful. For example, with the REST transport work, I can simply plug in gax and gapic-generator in build and generate, compile and run all unit tests for all APIs in both transport (27,000 tests in one run on one maching by simply replacing http_archive to local_repository). It is also very easy to do and repeat it after every change in the generator (and due to caching, the run is also fast). I literally have no idea how else I would test and fix this whole thing on googleapis scale without this bazel build.

**I'm afraid that we are on a path of fixing problems in release workflow by creating severe problems in development workflow**. The reason why the autogenerated PRs produce working and compilable code and Java, for example, does not suffer from the same problems as Python releases suffer (broken code being generated), is precisely due to gax and gapic-generator being tightly integrated into the bazel build.

Reply:
We fully intend to keep `dependencies.properties` up to date and we have no plans to break the Blaze build as part of the migration to Maven. If we are actually breaking any of the existing workflows, please let us know. However, this PR is not doing that.

Reply:
> I can simply plug in gax and gapic-generator in build and generate, compile and run all unit tests for all APIs in both transport

This is very interesting, can you please share the script you are using? I can make sure we are not breaking the script or create a similar experience with maven.

> The reason why the autogenerated PRs produce working and compilable code and Java, for example, does not suffer from the same problems as Python releases suffer (broken code being generated), is precisely due to gax and gapic-generator being tightly integrated into the bazel build.

Can you please elaborate the example? How bazel build can help prevent generating broken code? I thought it's mostly the code logic(like AST stuff) in the generator itself not the build system.